internal/engine/engine.go

```diff
@@ -154,24 +147,16 @@ func (c *craneEngine) ExecuteChecks(ctx context.Context) error {
 
 	// export/flatten, and extract
 	logger.V(log.DBG).Info("exporting and flattening image")
-	r, w := io.Pipe()
-	go func() {
-		logger.V(log.DBG).Info("writing container filesystem", "outputDirectory", containerFSPath)
-
-		// Close the writer with any errors encountered during
-		// extraction. These errors will be returned by the reader end
-		// on subsequent reads. If err == nil, the reader will return
-		// EOF.
-		w.CloseWithError(export(img, w))
-	}()
+	fs := mutate.Extract(img)
+	defer fs.Close()
```

Comment:
If I'm reading this correctly, this close doesn't complete until the end of preflight's entire run. Should we close this explicitly sooner in success cases, or otherwise put in a closure to close it  once the untar has completed?

Reply:
You are correct. Would be good to put in the closure to force the closure. :)